lib/ClusterShell/RangeSet.py

```diff
@@ -295,134 +318,128 @@ def __str__(self):
     # could be used to recreate a RangeSet with the same value
     __repr__ = __str__
 
-    def _contiguous_slices(self):
-        """Internal iterator over contiguous slices in RangeSet."""
-        k = j = None
-        for i in self._sorted():
-            if k is None:
-                k = j = i
-            if i - j > 1:
-                yield slice(k, j + 1, 1)
-                k = i
-            j = i
-        if k is not None:
-            yield slice(k, j + 1, 1)
+    def slices2(self, autostep=AUTOSTEP_DISABLED):
```

Comment:
to be eventually put in `slices()` ?

Reply:
I will see what I can do to merge `slices()` and `slices2()` :)